Index: l-smash-20140421/Makefile
===================================================================
--- l-smash-20140421.orig/Makefile
+++ l-smash-20140421/Makefile
@@ -43,6 +43,7 @@ install: all install-lib
 install-lib: liblsmash.pc lib
 	install -d $(DESTDIR)$(includedir)
 	install -m 644 $(SRCDIR)/lsmash.h $(DESTDIR)$(includedir)
+	install -m 644 $(SRCDIR)/importer/importer.h $(DESTDIR)$(includedir)/lsmash_importer.h
 	install -d $(DESTDIR)$(libdir)/pkgconfig
 	install -m 644 liblsmash.pc $(DESTDIR)$(libdir)/pkgconfig
 ifneq ($(STATICLIB),)
