Index: gpac-1.0.0/src/filters/dec_ac52.c
===================================================================
--- gpac-1.0.0.orig/src/filters/dec_ac52.c
+++ gpac-1.0.0/src/filters/dec_ac52.c
@@ -42,7 +42,6 @@
 #define uint8_t u8
 #endif
 
-#include <a52dec/mm_accel.h>
 #include <a52dec/a52.h>
 
 #define AC3_FRAME_SIZE 1536
@@ -78,11 +77,11 @@ static GF_Err a52dec_configure_pid(GF_Fi
 	ctx->ipid = pid;
 	gf_filter_pid_set_framing_mode(ctx->ipid, GF_TRUE);
 	if (!ctx->opid) {
-		u32 flags = MM_ACCEL_DJBFFT;
+//		u32 flags = MM_ACCEL_DJBFFT;
 #if !defined (GPAC_CONFIG_IOS) && !defined (GPAC_CONFIG_ANDROID)
-		flags |= MM_ACCEL_X86_MMX | MM_ACCEL_X86_3DNOW | MM_ACCEL_X86_MMXEXT;
+//		flags |= MM_ACCEL_X86_MMX | MM_ACCEL_X86_3DNOW | MM_ACCEL_X86_MMXEXT;
 #endif
-		ctx->codec = a52_init(flags);
+		ctx->codec = a52_init();
 		if (!ctx->codec) {
 			GF_LOG(GF_LOG_ERROR, GF_LOG_CODEC, ("[A52] Error initializing decoder\n"));
 			return GF_IO_ERR;
