Index: l-smash-2.14.5/configure
===================================================================
--- l-smash-2.14.5.orig/configure
+++ l-smash-2.14.5/configure
@@ -493,7 +493,7 @@ EOF
 for tool in $TOOLS; do
     cat >> config.mak2 << EOF
 cli/${tool}${EXT}: cli/${tool}.o $OBJ_TOOLS $STATICLIB $SHAREDLIB
-	\$(CC) \$(CFLAGS) \$(LDFLAGS) -o \$@ \$< $OBJ_TOOLS -llsmash \$(LIBS)
+	\$(CC) \$(CFLAGS) -fPIE \$(LDFLAGS) -pie -o \$@ \$< $OBJ_TOOLS -llsmash \$(LIBS)
 	-@ \$(if \$(STRIP), \$(STRIP) \$@)
 
 EOF
