---
 src/r8168.h   |    4 ++++
 src/r8168_n.c |   15 ++++++++++++++-
 2 files changed, 18 insertions(+), 1 deletion(-)

--- src/r8168.h
+++ src/r8168.h	2024-03-13 11:51:59.011947503 +0000
@@ -196,6 +196,10 @@ do { \
 } while (0)
 #endif //LINUX_VERSION_CODE < KERNEL_VERSION(2,6,15)
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0))
+#include <linux/sched/signal.h>
+#endif
+
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4,0,0)
 #if defined(skb_vlan_tag_present) && !defined(vlan_tx_tag_present)
 #define vlan_tx_tag_present skb_vlan_tag_present
--- src/r8168_n.c
+++ src/r8168_n.c	2024-03-13 11:57:02.890342590 +0000
@@ -61,11 +61,16 @@
 #include <linux/rtnetlink.h>
 #include <linux/completion.h>
 
+#ifndef HAS_PCI_ASPM_H
+#define HAS_PCI_ASPM_H @ASPM@
+#endif
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26)
 #if LINUX_VERSION_CODE < KERNEL_VERSION(5,4,0)
+#if HAS_PCI_ASPM_H
 #include <linux/pci-aspm.h>
 #endif
 #endif
+#endif
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,37)
 #include <linux/prefetch.h>
 #endif
@@ -586,8 +591,12 @@ static void rtl8168_set_rx_mode(struct n
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0)
 static void rtl8168_tx_timeout(struct net_device *dev, unsigned int txqueue);
 #else
+#if !HAS_PCI_ASPM_H
+static void rtl8168_tx_timeout(struct net_device *dev, unsigned int txqueue);
+#else
 static void rtl8168_tx_timeout(struct net_device *dev);
 #endif
+#endif
 static struct net_device_stats *rtl8168_get_stats(struct net_device *dev);
 static int rtl8168_rx_interrupt(struct net_device *, struct rtl8168_private *, struct rtl8168_rx_ring *, napi_budget);
 #ifdef CONFIG_R8168_NAPI
@@ -30897,8 +30906,12 @@ static void
 rtl8168_tx_timeout(struct net_device *dev, unsigned int txqueue)
 #else
 static void
+#if !HAS_PCI_ASPM_H
+rtl8168_tx_timeout(struct net_device *dev, unsigned int txqueue)
+#else
 rtl8168_tx_timeout(struct net_device *dev)
 #endif
+#endif
 {
         struct rtl8168_private *tp = netdev_priv(dev);
 
@@ -31636,7 +31649,7 @@ rtl8168_rx_interrupt(struct net_device *
 
                 if (rtl8168_rx_vlan_skb(tp, desc, skb) < 0)
                         rtl8168_rx_skb(tp, skb, ring_index);
-#if LINUX_VERSION_CODE < KERNEL_VERSION(4,11,0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,4,0)
                 dev->last_rx = jiffies;
 #endif //LINUX_VERSION_CODE < KERNEL_VERSION(4,11,0)
                 RTLDEV->stats.rx_bytes += pkt_size;
