Index: libopenaptx-0.2.0/Makefile
===================================================================
--- libopenaptx-0.2.0.orig/Makefile
+++ libopenaptx-0.2.0/Makefile
@@ -37,7 +37,7 @@ SOURCES = $(NAME).c
 AOBJECTS = $(NAME).o
 IOBJECTS = $(NAME)enc.o $(NAME)dec.o
 
-BUILD = $(SOFILENAME) $(SONAME) $(LIBNAME) $(ANAME) $(AOBJECTS) $(IOBJECTS) $(UTILITIES) $(STATIC_UTILITIES)
+BUILD = $(SOFILENAME) $(SONAME) $(LIBNAME) $(IOBJECTS) $(UTILITIES)
 
 all: $(BUILD)
 
@@ -46,7 +46,7 @@ clean:
 
 install: $(BUILD)
 	$(MKDIR) $(DESTDIR)$(PREFIX)/$(LIBDIR)
-	$(CP) $(SOFILENAME) $(SONAME) $(LIBNAME) $(ANAME) $(DESTDIR)$(PREFIX)/$(LIBDIR)
+	$(CP) $(SOFILENAME) $(SONAME) $(LIBNAME) $(DESTDIR)$(PREFIX)/$(LIBDIR)
 	$(MKDIR) $(DESTDIR)$(PREFIX)/$(BINDIR)
 	$(CP) $(UTILITIES) $(DESTDIR)$(PREFIX)/$(BINDIR)
 	$(MKDIR) $(DESTDIR)$(PREFIX)/$(INCDIR)
