Index: kvazaar-2.3.2/CMakeLists.txt
===================================================================
--- kvazaar-2.3.2.orig/CMakeLists.txt
+++ kvazaar-2.3.2/CMakeLists.txt
@@ -258,7 +258,7 @@ else()
   # CYGWIN, MSYS, and MINGW seem to be needing this but in some cases 
   # it might be that the toolset is not properly set, so also use this
   # in cases where we are not sure that it is not needed
-  if((NOT MSVC AND NOT LINUX AND NOT APPLE AND NOT ANDROID AND NOT EMSCRIPTEN AND NOT BSD) OR (CYGWIN OR MSYS OR MINGW))
+  if(FALSE)
     set(CMAKE_C_FLAGS "-Wa,-muse-unaligned-vector-move ${CMAKE_C_FLAGS}")
   endif()
 
