From: Sebastian Ramacher <sramacher@debian.org>
Date: Fri, 31 May 2024 23:38:23 +0200
Subject: Do not build with assembly support on arm*

---
 source/CMakeLists.txt |    9 ---------
 1 file changed, 9 deletions(-)

--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -72,15 +72,6 @@
         add_definitions(-DPPC64=1)
         message(STATUS "Detected POWER PPC64 target processor")
     endif()
-elseif(ARMMATCH GREATER "-1")
-    if(CROSS_COMPILE_ARM)
-        message(STATUS "Cross compiling for ARM arch")
-    else()
-        set(CROSS_COMPILE_ARM 0)
-    endif()
-	message(STATUS "Detected ARM target processor")
-    set(ARM 1)
-    add_definitions(-DX265_ARCH_ARM=1 -DHAVE_ARMV6=1)
 elseif(ARM64MATCH GREATER "-1")
     message(STATUS "Detected ARM64 target processor")
     set(ARM64 1)
