diff -rup a/applications/generators/SVG/svggen.h b/applications/generators/SVG/svggen.h
--- a/applications/generators/SVG/svggen.h	2024-09-20 12:10:23.374809464 +0200
+++ b/applications/generators/SVG/svggen.h	2024-09-20 12:30:06.268430563 +0200
@@ -190,6 +190,17 @@ static _atts generic_attributes[] = {
 
 FILE *BeginFile(u32 type);
 void EndFile(FILE *f, u32 type);
+SVGGenAttribute *findAttribute(SVGGenElement *e, char *name);
+void generate_laser_tables(GF_List *svg_elements);
+void generate_laser_tables_da(GF_List *atts);
+void generate_table(GF_List *elements);
+void generateAttributes(FILE *output, GF_List *attributes, Bool inDefine);
+u32 generateCoreInfo(FILE *output, SVGGenElement *elt, u32 start);
+u32 generateGenericInfo(FILE *output, SVGGenElement *elt, u32 index, char *pointer_root, u32 start);
+void generateSVGCode_V1(GF_List *svg_elements);
+void generateSVGCode_V2(GF_List *svg_elements);
+void generateSVGCode_V3(GF_List *svg_elements);
+void svgNameToImplementationName(xmlChar *svg_name, char implementation_name[50]);
 
 
 #endif // _SVGGEN_H_
