--- rtmpsrv.c.orig	2010-04-03 01:51:12.000000000 +0200
+++ rtmpsrv.c	2010-04-03 01:51:25.000000000 +0200
@@ -39,6 +39,7 @@
 #include "thread.h"
 
 #ifdef linux
+#include <linux/types.h>
 #include <linux/netfilter_ipv4.h>
 #endif
 
--- rtmpsuck.c.orig	2010-04-03 01:51:33.000000000 +0200
+++ rtmpsuck.c	2010-04-03 01:51:44.000000000 +0200
@@ -39,6 +39,7 @@
 #include "thread.h"
 
 #ifdef linux
+#include <linux/types.h>
 #include <linux/netfilter_ipv4.h>
 #endif
 
