diff -rup a/Source/CTest/cmCTestCurl.cxx b/Source/CTest/cmCTestCurl.cxx
--- a/Source/CTest/cmCTestCurl.cxx	2025-04-10 15:13:20.000000000 +0200
+++ b/Source/CTest/cmCTestCurl.cxx	2025-10-05 10:01:30.991768658 +0200
@@ -264,7 +264,7 @@ void cmCTestCurl::SetProxyType()
 {
   this->HTTPProxy.clear();
   // this is the default
-  this->HTTPProxyType = CURLPROXY_HTTP;
+  this->HTTPProxyType = (curl_proxytype) CURLPROXY_HTTP;
   this->HTTPProxyAuth.clear();
   if (cmSystemTools::GetEnv("HTTP_PROXY", this->HTTPProxy)) {
     std::string port;
@@ -276,11 +276,11 @@ void cmCTestCurl::SetProxyType()
     if (cmSystemTools::GetEnv("HTTP_PROXY_TYPE", type)) {
       // HTTP/SOCKS4/SOCKS5
       if (type == "HTTP") {
-        this->HTTPProxyType = CURLPROXY_HTTP;
+        this->HTTPProxyType = (curl_proxytype) CURLPROXY_HTTP;
       } else if (type == "SOCKS4") {
-        this->HTTPProxyType = CURLPROXY_SOCKS4;
+        this->HTTPProxyType = (curl_proxytype) CURLPROXY_SOCKS4;
       } else if (type == "SOCKS5") {
-        this->HTTPProxyType = CURLPROXY_SOCKS5;
+        this->HTTPProxyType = (curl_proxytype) CURLPROXY_SOCKS5;
       }
     }
     cmSystemTools::GetEnv("HTTP_PROXY_USER", this->HTTPProxyAuth);
